File: app/src/main/java/com/boardgamegeek/sorter/MyRatingSorter.java
package com.boardgamegeek.sorter;

import android.content.Context;
import android.support.annotation.NonNull;
import android.support.annotation.StringRes;

import com.boardgamegeek.R;
import com.boardgamegeek.provider.BggContract.Collection;

import java.text.DecimalFormat;

public class MyRatingSorter extends RatingSorter {
	private static final DecimalFormat DISPLAY_FORMAT = new DecimalFormat("0.0");

	public MyRatingSorter(@NonNull Context context) {
		super(context);
	}

	@StringRes
	@Override
	protected int getDescriptionId() {
		return R.string.collection_sort_my_rating;
	}

	@StringRes
	@Override
	public int getTypeResource() {
		return R.string.collection_sort_type_my_rating;
	}

	@Override
	protected String getSortColumn() {
		return Collection.RATING;
	}


	@Override
	protected DecimalFormat getDisplayFormat() {
		return DISPLAY_FORMAT;
	}
}

File: app/src/main/java/com/boardgamegeek/sorter/MyRatingSorter.kt
package com.boardgamegeek.sorter

import android.content.Context
import androidx.annotation.StringRes
import com.boardgamegeek.R
import com.boardgamegeek.entities.CollectionItem
import com.boardgamegeek.extensions.asPersonalRating
import java.text.DecimalFormat

class MyRatingSorter(context: Context) : RatingSorter(context) {
    override val ascendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_my_rating_asc

    override val descendingSortTypeResId: Int
        @StringRes
        get() = R.string.collection_sort_type_my_rating

    override val descriptionResId: Int
        @StringRes
        get() = R.string.collection_sort_my_rating

    override fun sortAscending(items: Iterable<CollectionItem>) = items.sortedBy { it.rating }

    override fun sortDescending(items: Iterable<CollectionItem>) = items.sortedByDescending { it.rating }

    override val displayFormat = DecimalFormat("0.0")

    override fun getRating(item: CollectionItem) = item.rating

    override fun getRatingText(item: CollectionItem) = getRating(item).asPersonalRating(context)
}
